### PR TITLE
Update 3.1.0

### PR DESCRIPTION
**Pull request by Arishamays1**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ requirements:
   run:
     - python
     - alembic >=1.4
+    - importlib-metadata >=3.6
```
Comment (cbouss) [inline recipe/meta.yaml]: This should be for # [py < 310] (see https://github.com/jupyterhub/jupyterhub/blob/3.1.0/requirements.txt#L4)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -41,11 +42,13 @@ requirements:
     - pycurl
```
Comment (cbouss) [inline recipe/meta.yaml]: psutil <= 5.6.5 (see https://github.com/jupyterhub/jupyterhub/blob/3.1.0/requirements.txt#L11)

File: recipe/meta.yaml [hunk 3]
```diff
@@ -41,11 +42,13 @@ requirements:
...
     - python-dateutil
     - requests
+    - packaging
     - sqlalchemy >=1.1
     - tornado >=5.1
     - traitlets >=4.3.2
     - jupyter_telemetry >=0.1.0
-
+  run_constrained:
```
Comment (cbouss) [inline recipe/meta.yaml]: This could be removed as it is covered in the run section.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -41,11 +42,13 @@ requirements:
     - pycurl
     - python-dateutil
     - requests
+    - packaging
     - sqlalchemy >=1.1
     - tornado >=5.1
     - traitlets >=4.3.2
     - jupyter_telemetry >=0.1.0
```
Comment (cbouss) [inline recipe/meta.yaml:47]: I think the following can be removed. At least I don't see mention of them in setup.py nor requirements.txt.
- entrypoints (repalced by importlib-metadata)
- pycurl

File: recipe/meta.yaml (hunk 2 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Made the edit to psutil >=5.6.5. Thanks!

File: recipe/meta.yaml (hunk 3 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Great, removed

File: recipe/meta.yaml (hunk 4 above)
Comment (Arishamays1) [reply recipe/meta.yaml:47]: Got it thanks! Removed

File: recipe/meta.yaml (hunk 1 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Thanks, Charles. Could you clarify what changes are needed for this in particular?

Comment (cbouss) [reply recipe/meta.yaml]: Sure, this line:
`importlib_metadata>=3.6; python_version < '3.10'`
Translates to this in a conda recipe:
`- importlib_metadata >=3.6. # [py<310]`

**Review by varlackc — COMMENTED** [review]
After the requested changes are made to importlib-metadata we might be able to approve

we need to have the following

`- importlib-metadata >=3.6  # [py<310]`

https://github.com/AnacondaRecipes/jupyterhub-feedstock/blob/62eb3b8ec6b158eb0bb03405aa1e93def9d6c0b6/recipe/meta.yaml#L32

File: recipe/meta.yaml (hunk 1 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: @cbouss @varlackc It seems that changes to the pinning for this dependency causes test failures and therefore build failures for all archetypes. See [here.](https://github.com/AnacondaRecipes/jupyterhub-feedstock/pull/15/commits/2314bb248f0f112d7b3c8dbde151dc4ec4c4dfbc) 
This change returns a `Module not found` error for `importlib_metadata`. Would reconsidering or modifying this pinning be an option?

Comment (cbouss) [reply recipe/meta.yaml]: I noticed I had a typo in my previous comment. An extra dot.
This [commit](8573325e33f34567612d8685f6e5e46c3a8bd6e6) had that typo.
The next [commit](2314bb248f0f112d7b3c8dbde151dc4ec4c4dfbc) had the wrong selector.

This works:
`- importlib_metadata >=3.6  # [py<310]`

Comment (Arishamays1) [reply recipe/meta.yaml]: Awesome Thanks so much! This change has been added.

**Review by cbouss — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**